modules/objdetect/include/opencv2/objdetect/aruco_dictionary.hpp

```diff
@@ -0,0 +1,179 @@
+ */
+class CV_EXPORTS_W Dictionary {
+
+    public:
+    CV_PROP_RW Mat bytesList;         // marker code information
+    CV_PROP_RW int markerSize;        // number of bits per dimension
+    CV_PROP_RW int maxCorrectionBits; // maximum number of bits that can be corrected
+
+
+    Dictionary(const Mat &bytesList = Mat(), int _markerSize = 0, int maxcorr = 0);
+
+
+    Dictionary(const Ptr<Dictionary> &dictionary);
+
+
+    /** @brief returns generateCustomDictionary(nMarkers, markerSize, randomSeed)
+     *
+     * @see generateCustomDictionary
+     */
+    CV_WRAP_AS(create) static Ptr<Dictionary> create(int nMarkers, int markerSize, int randomSeed=0);
+
+
+    /** @brief returns generateCustomDictionary(nMarkers, markerSize, baseDictionary, randomSeed)
+     *
+     * @see generateCustomDictionary
+     */
+    CV_WRAP_AS(create_from) static Ptr<Dictionary> create(int nMarkers, int markerSize,
+            const Ptr<Dictionary> &baseDictionary, int randomSeed=0);
+
+    /** @brief Read a new dictionary from FileNode.
+     *
+     * Dictionary format:\n
+     * nmarkers: 35\n
+     * markersize: 6\n
+     * maxCorrectionBits: 5\n
+     * marker_0: "101011111011111001001001101100000000"\n
+     * ...\n
+     * marker_34: "011111010000111011111110110101100101"
+     */
+    CV_WRAP bool readDictionary(const cv::FileNode& fn);
+
+    /** @brief Write a dictionary to FileStorage, format is the same as in readDictionary().
+     */
+    CV_WRAP void writeDictionary(Ptr<FileStorage>& fs);
+
+    /**
+     * @see getPredefinedDictionary
+     */
+    CV_WRAP static Ptr<Dictionary> get(int dict);
+
+    /** @brief Given a matrix of bits. Returns whether if marker is identified or not.
+     *
+     * It returns by reference the correct id (if any) and the correct rotation
+     */
+    CV_WRAP bool identify(const Mat &onlyBits, CV_OUT int &idx, CV_OUT int &rotation, double maxCorrectionRate) const;
+
+    /** @brief Returns the distance of the input bits to the specific id.
+     *
+     * If allRotations is true, the four posible bits rotation are considered
+     */
+    CV_WRAP int getDistanceToId(InputArray bits, int id, bool allRotations = true) const;
+
+
+    /** @brief Draw a canonical marker image
+     */
+    CV_WRAP void drawMarker(int id, int sidePixels, OutputArray _img, int borderBits = 1) const;
+
+
+    /** @brief Transform matrix of bits to list of bytes in the 4 rotations
+      */
+    CV_WRAP static Mat getByteListFromBits(const Mat &bits);
+
+
+    /** @brief Transform list of bytes to matrix of bits
+      */
+    CV_WRAP static Mat getBitsFromByteList(const Mat &byteList, int markerSize);
+};
+
+
+
+
+/** @brief Predefined markers dictionaries/sets
+ *
+ * Each dictionary indicates the number of bits and the number of markers contained
+ * - DICT_ARUCO_ORIGINAL: standard ArUco Library Markers. 1024 markers, 5x5 bits, 0 minimum
+                          distance
+ */
+enum PREDEFINED_DICTIONARY_NAME {
```

Comment:
Camel-case naming.

Reply:
ping

Reply:
rename `PREDEFINED_DICTIONARY_NAME` to `PREDEFINED_DICTIONARY`

Reply:
It should be `PredefinedDictionary` by coding style.
`PredefinedDictionaryType` by descriptive naming.

Reply:
rename `PREDEFINED_DICTIONARY` to `PredefinedDictionaryType`